lib/cisco_node_utils/bridge_domain.rb

```diff
@@ -0,0 +1,247 @@
+# February 2016, Rohan Gandhi Korlepara
+#
+# Copyright (c) 2015-2016 Cisco and/or its affiliates.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+require_relative 'node_util'
+require_relative 'feature'
+require_relative 'vdc'
+
+module Cisco
+  # node_utils class for bridge_domain
+  class BridgeDomain < NodeUtil
+    attr_reader :name, :bd_ids
+
+    def initialize(bds, instantiate=true)
+      @bd_ids = bds.to_s
+      fail 'bridge-domain value is empty' if @bd_ids.empty? # no empty strings
+      fail 'bridge-domain value has spaces' if @bd_ids.include?(' ') # no spaces
+
+      # Check if the given input bd_ids does contain only numeric value or range
+      # of numeric values
+      # Also bd_ids need to fall in range of 2-2967
+      narray = @bd_ids.split(',')
+      narray.each do |elem|
+        if elem.include?('-')
+          earray = elem.split('-')
+          earray.each do |temp_elem|
+            fail 'bridge-domain cannot be a string' unless
+                                    BridgeDomain.number?(temp_elem)
+            fail 'Invalid bridge-domain value' unless
+                                    temp_elem.to_i > 1 && temp_elem.to_i < 3968
+          end
+        else
+          fail 'bridge-domain cannot be a string' unless
+                                    BridgeDomain.number?(elem)
+          fail 'Invalid bridge-domain value' unless
+                                    elem.to_i > 1 && elem.to_i < 3968
+        end
+      end
+
+      create if instantiate
+    end
+
+    def self.number?(string)
+      true if Integer(string)
+    rescue
+      false
+    end
+
+    # This will expand the string to a list of bds as integers
+    def self.bd_ids_to_array(bdid_string)
+      list = []
+      narray = bdid_string.split(',')
+      narray.each do |elem|
+        if elem.include?('-')
+          es = elem.gsub('-', '..')
+          ea = es.split('..').map { |d| Integer(d) }
+          er = ea[0]..ea[1]
+          list << er.to_a
+        else
+          list << elem.to_i
+        end
+      end
+      list.flatten
+    end
+
+    # This method will generate a batched string if a list is passed as
+    # argument
+    # Input would be as [1,2,3,4,5,10,11,12,7,30,100,31,32]
+    # output will be 1-5,10-12,7,30,100,31-32
```

Comment:
Yes ordering is important as this is used in member_vni getter and if the vni mapping are not in order then it would cause an issue.
Eg.
bridge-domain 100,105,108-110,150
   member vni 5000, 8000, 5008-5009, 7000, 5050